guava/src/com/google/common/escape/UnicodeEscaper.java

```diff
@@ -154,13 +162,24 @@ protected int nextEscapeIndex(CharSequence csq, int start, int end) {
    * @throws NullPointerException if {@code string} is null
    * @throws IllegalArgumentException if invalid surrogate characters are encountered
    */
-  protected final String escapeSlow(String s, int index) {
+  @SuppressWarnings(value = {"upperbound:compound.assignment.type.incompatible",/*
+          (1): `destIndex` is always @LTEqLengthOf("dest") @LessThan("destSize + 1") because `dest` array
+          will always be regrow when `destSize` is less than `sizeNeeded` */
+          "upperbound:argument.type.incompatible",/*
+          (2): Because of System.arraycopy() method, `escaped.length` is required to be
+          @LTLengthOf(value={"escaped", "dest"}, offset={"-1", "destIndex - 1"}).
+          Since escaped.length is length of `escaped`, it should already be inferred to have length of @LTLengthOf(value="escaped", offset="-1")
```

Comment:
This comment doesn't explain why `escaped.length` is safe to use in this call to System.arraycopy - why does it have the appropriate relation to the offset?